File: 2379 minimum-recolors-to-get-k-consecutive-black-blocks/solution.go
func minimumRecolors(blocks string, k int) int {
    count:=0
    for i:=0;i<k;i++ {
        if blocks[i]=='W' {
            count++
        }
    }
    minn:=count
    for i:=k;i<len(blocks);i++ {
        if blocks[i]=='W' {
            count++
        }
        if blocks[i-k]=='W' {
            count--
        }
        minn=min(minn,count)
    }
    return minn
}

File: 2379 minimum-recolors-to-get-k-consecutive-black-blocks/solution.rs
impl Solution {
    pub fn minimum_recolors(blocks: String, k: i32) -> i32 {
        let k=k as usize;
        let bytes = blocks.as_bytes();
        let mut count=0;
        for i in 0..k {
             if bytes[i]==b'W' {
            count+=1;
            }
        }
        let mut mi=count;
        for i in k..blocks.len() {
            if bytes[i]==b'W' {
            count+=1;
            }
            if bytes[i-k]==b'W' {
            count-=1;
            }
            if  count<mi {
            mi=count;
            }
        }
        mi
    }
}
